File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int kthSmallest(struct TreeNode* root, int k) {
    void preorder(struct TreeNode* root,int arr[],int *a){
        if(root){
            preorder(root->left,arr,a);
            arr[(*a)++]=root->val;
            preorder(root->right,arr,a);
        }
    }
    int arr[100000];
    int a=0;
    preorder(root,arr,&a);
    return arr[k-1];
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        stack=[]
        def inorder(root):
            if root:
                inorder(root.left)
                stack.append(root.val)
                inorder(root.right)
        inorder(root)
        while k:
            a=stack.pop(0)
            k-=1
        return a            